theano/scan_module/scan_perform.pyx

```diff
@@ -514,7 +514,8 @@ def perform(
             if i == 0:
                 jout = j+offset_out
                 shape = (store_steps[j],) + output_storage[jout].storage[0].shape
-                if len(output_storage[jout].storage[0].shape) == 0:
+                if output_storage[jout].storage[0].ndim == 0 and \
+                    isinstance(output_storage[jout].storage[0], numpy.ndarray):
```

Comment:
One question that you can't probably answer, why do we modify vector_outs here? This is a permanent change for all futur call to this Theano function. The previous condition was on the current shape (even if constant for all the iteration of the current Theano function call, I'm pretty sure futur Theano function call could have a different shape).

Your new tests on ndim is better then the old on the shapes. But I still don't understand why it change vector_outs here and why it wasn't set correctly around line 616 in this file. Do you have an idea?